src/sqlfluff/utils/reflow/respace.py

```diff
@@ -29,47 +29,77 @@ def determine_constraints(
     """Given the surrounding blocks, determine appropriate constraints."""
     # Start with the defaults.
     pre_constraint = prev_block.spacing_after if prev_block else "single"
+    # Unless align, split.
+    if pre_constraint.startswith("align"):
+        modifier = ""
+    else:
+        pre_constraint, _, modifier = pre_constraint.partition(":")
+    if not modifier:
+        pass
+    elif modifier == "inline":
+        strip_newlines = True
+    else:  # pragma: no cover
+        raise NotImplementedError(
+            f"Unexpected within constraint modifier: {pre_constraint}"
+        )
+
     post_constraint = next_block.spacing_before if next_block else "single"
+    # Unless align, split.
+    if post_constraint.startswith("align"):
```

Comment:
Can you not just loop through the constraints rather than repeating the code block?

Reply:
yeah that's probably a better idea... 🤔